src/puppeteer_utils.js

```diff
@@ -256,6 +256,116 @@ const cacheResponses = async (page, cache, responseUrlRules) => {
     });
 };
 
+/**
+ * Setup which requests done by the page should be block and which responses should be cached.
+ * This helper function exists mainly because it's not possible to attach two handler's that both intercept the same request.
+ * *IMPORTANT*: Caching responses stores them to memory, so too loose rules could cause memory leaks for longer running crawlers.
+ *   This issue should be resolved or atleast mitigated in future iterations of this feature.
+ * String rules are compared as request().url().includes(rule) while RegExp rules are evaluated as rule.test(request.url()).
```

Comment:
Please enclose all code comments to "`" brackets, so they are rendered as code in the docs.